activerecord/lib/active_record/railtie.rb

```diff
@@ -36,6 +36,13 @@ class Railtie < Rails::Railtie # :nodoc:
     config.eager_load_namespaces << ActiveRecord
 
     rake_tasks do
+      task :test do
+        if !ENV['RUN_AGAINST_PRODUCTION_DATABASE'] && ActiveRecord::Migrator.production?
```

Comment:
If you accept my previous comment, we should probably rename this to something like `DISABLE_DATABASE_ENVIRONMENT_CHECK`